docs/reference/performance.md

```diff
@@ -0,0 +1,212 @@
+3. Install Gatekeeper and accompanying `Constraint` and `ConstraintTemplate`.
+4. Install Ratify with test-specific configuration
+5. Install Prometheus
+6. Generate a set of workload resource templates depending on the test configuration. Specifically, generate templates based on number of containers and image references with attached artifacts.
+7. Begin `ResourceUsage` measurements to understand resource consumption as resources are created in cluster.
+8. Apply test workloads and wait for deployment to succeed. After, destroy the resources.
+9. Scrape Ratify metrics using prometheus
+10. Upload logs for Gatekeeper and Ratify. Upload resource consumption report.
+
+### Variables
+
+#### Cluster Size
+The node count of the cluster needs to scale according to the workload size. The test conducted imposed a 40 pod scheduler limit per node.
+
+Workload pod counts: 200, 2000, 10000
+Cluster node counts: 11, 62, 307
+Node Type: Standard_D4s_v3 4vCPUs 16Gib memory. Details [here](https://learn.microsoft.com/en-us/azure/virtual-machines/dv3-dsv3-series#dsv3-series)
+
+#### Registry Region Size
+
+ACR has different region sizes. The size of the region determines the max RPS for a single ACR from a single IP. The test focuses on 4 regions of varying sizes. 
+
+- East US 2: Large
+- East US 2 Availability-Zone-enabled: Medium
+- Central US Availability-Zone-enabled: Small
+- East US 2 Canary: Extra Small
+
+ACR regions were selected by size and proximity to the cluster region of East US 2 to minimize cross-region latency. 
+
+#### Workload Resource Type
+
+The test focused on Deployment and Job workloads. Most tests were conducted with a single Deployment with varying replica counts. Job tests applied 5 Jobs simultaneously splitting total pod count evenly across each job. 
+
+#### Distributed caching
+Majority of tests enabled distributed caching across 3 replicas for optimal performance configuration. A subset tested Ratify with 3 replicas with the distributed cache disabled. Distributed caching is critical for reducing duplicate registry requests. It is also essential for external plugins utilizing caching.
+
+Test enabled/disabled RATIFY_EXPERIMENTAL_HIGH_AVAILABILITY feature flag
+
+#### Ratify Single Instance
+This test aimed to determine the workload threshold for a single instance of Ratify before over-exhaustion. Single instance vs 3 Ratify Replicas.
+
+
+### Measurements
+
+1. Registry Throttling
+    - \# of 429 http responses returned from registry
+    - This indicates supported registry RPS in a particular region.
+1. Ratify pod(s) CPU/memory usage: High Memory consumption is correlated with high RPS load on Ratify.
+1. Approximate max verification request duration: Helpful to understand initial non-cached request durations
+3. P95 verification request duration: Shows distribution of request durations through duration of the test. 
+4. Approximate max mutation request duration: Helpful to understand initial non-cache request durations.
```

Comment:
Wonder if we could add p99 in the future.

Reply:
Yes that's a good point. I actually did consider p99 however p99 is not as different as p95. With large scale testing, the request # is so large that the distribution of latencies is extremely narrow. P95 vs P99 is almost the same. Initial verification latency on first request magnitudes larger than rest of the requests that we can only really capture that via a max metric.